src/Exception/MutatorException.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Exception;
+
+use Infection\Mutator\Util\Mutator;
+
+/**
+ * @internal
+ */
+final class MutatorException extends \Exception
```

Comment:
What about calling this class `InvalidMutator` instead of `internalErrorWhileTraversing()` `create()`? IMO this is more readable and explicit:

```php
throw InvalidMutator::create();
```

than:

```php
throw MutatorException:: internalErrorWhileTraversing();
```